### PR TITLE
multi-gpu batch inference available

### PR DESCRIPTION
**Pull request by yaoguany**

### CODE REVIEW

**Review by research4pan — COMMENTED** [review]
Looks good to me. If all tests passed after `git rebase`, then we can merge this pull request. Thanks @yaoguany . Some minor problems, but can be improved later.

### `src/lmflow/models/hf_decoder_model.py`

- [Style] line 345: docstring here can be better, e.g. '' -> ", ' -> ", attention_masks can be specified in the example as well.
-

File: src/lmflow/models/hf_decoder_model.py [hunk 1]
```diff
@@ -226,13 +227,14 @@ def __init__(
...
                 )
 
-            self.tokenizer = AutoTokenizer.from_pretrained(model_args.model_name_or_path)
             self.backend_model_full = self.backend_model
             if peft_model_id is not None:
                 self.backend_model = PeftModel.from_pretrained(
                     self.backend_model, peft_model_id
                 )
-
+
```
Comment (shizhediao) [inline src/lmflow/models/hf_decoder_model.py:235]: #necessary for llama,gpt2 and other decoder model

should be 
#necessary for llama, gpt2 and other decoder model

File: src/lmflow/models/hf_decoder_model.py [hunk 2]
```diff
@@ -241,6 +243,9 @@ def __init__(
         elif tune_strategy == 'adapter':
             raise NotImplementedError('adapter tune strategy not implemented')
 
+        if self.tokenizer.eos_token_id is None:
+            self.tokenizer.eos_token_id = self.backend_model.config.eos_token_id
+        self.tokenizer.pad_token_id = self.tokenizer.eos_token_id
```
Comment (shizhediao) [inline src/lmflow/models/hf_decoder_model.py:248]: is it a must to set pad_token_id as eos_token_id?
what if the model has a  special pad_token_id originally?

Could we add `if self.tokenizer.pad_token_ids is None`

**Review by shizhediao — APPROVED** [review]
LGTM!

**Pull request closed without merging**